AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/editor_script_canvas.py

```diff
@@ -37,53 +36,46 @@ class VariableState(Enum):
 class ScriptCanvasComponent:
 
     def __init__(self):
-        self.editor_entity = None
         self.script_canvas_component = None
 
-    def __init__(self, editor_entity: EditorEntity, sc_file_path: str):
-        self.editor_entity = None
-        self.script_canvas_component = None
-        self.add_component_to_entity(editor_entity, sc_file_path)
+    def get_script_canvas_component(self) -> EditorComponent:
+        return self.script_canvas_component
 
-    def create_new_entity_with_component(self, entity_name: str, sc_file_path: str,
-                                         position=math.Vector3(512.0, 512.0, 32.0)) -> None:
+    def add_component_to_existing_entity(self, editor_entity: EditorEntity, sc_file_path: str) -> None:
         """
-        Function for constructing the SC Component object. This will make a new entity, add a script canvas component and
-        then open a script canvas file from disk onto the component's source handle value.
+        Function for initializing a script canvas component object if you already have an editor entity you want to use.
 
-        param entity_name: The name you want the entity to have
-        param sc_file_path: The path on disk to the script canvas file
-        param position: optional parameter for the position of the entity
+        param editor_entity: The entity we want to configure
+        param sc_file_path: location on disk to the script canvas file
 
-        Returns None
+        returns None
         """
         sourcehandle = scriptcanvas.SourceHandleFromPath(sc_file_path)
 
-        self.editor_entity = EditorEntity.create_editor_entity_at(position, entity_name)
-        self.editor_entity.add_components([SCRIPT_CANVAS_UI])
-
-        self.script_canvas_component = self.editor_entity.get_components_of_type([SCRIPT_CANVAS_UI])[0]
+        self.script_canvas_component = editor_entity.add_component(SCRIPT_CANVAS_UI)
```

Comment:
Unneeded. 

The pattern used here is for Editor Component Behavior classes to be created by the `EditorEntity` and the returned `EditorComponent` to be consumed and set in the initializer. We should not be doing it the other way around as the logic puts the onus of entity management on class that does not manipulate the entity directly.

Reply:
removed this function and updated the constructor to require an editor entity to get the component object.